### PR TITLE
Update `license header` and `scripts`

### PR DESCRIPTION
**Pull request by niklasad1**
Sorry for a huge `PR` but most of the changes are caused by updated scripts.

The major changes in this PR are:

1. Updated `scripts/add_license.sh` it will now replace the existing header it starts with `// Copyright` and ends with `If not, see <http://www.gnu.org/licenses/>.`  or just add the header if no header is found
2. Added a new script that supresses duplicated empty lines 

So, the most of important parts of this PR to review are:

1. [scripts/add_license.sh](https://github.com/niklasad1/parity/blob/license-header/update/scripts/add_license.sh)
2. [scripts/remove_duplicate_empty_lines.sh](https://github.com/niklasad1/parity/blob/license-header/update/scripts/remove_duplicate_empty_lines.sh)

### CODE REVIEW

File: dapps/js-glue/build.rs [hunk 1]
```diff
@@ -8,12 +8,11 @@
 
 // Parity is distributed in the hope that it will be useful,
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```
Comment (sorpaas) [inline dapps/js-glue/build.rs]: Regarding the extra space, I think it's actually intentional? Reading from [Wikipedia](https://en.wikipedia.org/wiki/Sentence_spacing) this is something called "English spacing". And I do remember encountering of this spacing strategy in my life.

And in particular, the original GPL texts uses English spacing. https://www.gnu.org/licenses/gpl.txt

File: ethcore/vm/src/return_data.rs [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
 // Parity is free software: you can redistribute it and/or modify
```
Comment (5chdn) [inline ethcore/vm/src/return_data.rs]: Please check what caused this to have two headers.

File: rpc/src/v1/helpers/network_settings.rs [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
 // Copyright 2015-2017 Parity Technologies (UK) Ltd.
 // This file is part of Parity.
```
Comment (5chdn) [inline rpc/src/v1/helpers/network_settings.rs]: Check double header.

**Comment by 5chdn** [comment]
This PR kills my browser. I reported this to Github. I'll try to continue reviewing...

**Comment by dvdplm** [comment]
I noticed we have a few `Cargo.toml` that are missing the `license` key. Not sure if now is a good time to add those.

e.g.
`ethcore/res/wasm-tests/gen/Cargo.toml`
`ethcore/wasm/run/Cargo.toml`
`ethstore/cli/Cargo.toml`
`util/mem/Cargo.toml`
`util/stats/Cargo.toml`
`util/trie-standardmap/Cargo.toml`
etc

**Comment by niklasad1** [comment]
Good point, I found that the following `Cargo.tomls` were missing `license = "xxx"`

```bash
./evmbin/Cargo.toml
./local-store/Cargo.toml
./ethash/Cargo.toml
./json/Cargo.toml
./util/kvdb/Cargo.toml
./util/rlp_derive/Cargo.toml
./util/using_queue/Cargo.toml
./util/error/Cargo.toml
./util/macros/Cargo.toml
./util/rlp_compress/Cargo.toml
./util/trie-standardmap/Cargo.toml
./util/kvdb-rocksdb/Cargo.toml
./util/mem/Cargo.toml
./util/path/Cargo.toml
./util/kvdb-memorydb/Cargo.toml
./util/dir/Cargo.toml
./util/version/Cargo.toml
./util/stats/Cargo.toml
./util/stop-guard/Cargo.toml
./util/migration-rocksdb/Cargo.toml
./util/unexpected/Cargo.toml
./ethkey/cli/Cargo.toml
./ethkey/Cargo.toml
./chainspec/Cargo.toml
./ethcore/service/Cargo.toml
./ethcore/evm/Cargo.toml
./ethcore/wasm/run/Cargo.toml
./ethcore/wasm/Cargo.toml
./ethcore/types/Cargo.toml
./ethcore/transaction/Cargo.toml
./ethcore/crypto/Cargo.toml
./ethcore/vm/Cargo.toml
./ethcore/res/wasm-tests/gen/Cargo.toml
./machine/Cargo.toml
./ethstore/cli/Cargo.toml
./ethstore/Cargo.toml
./whisper/Cargo.toml
```
However, I'm not sure if we should just append `license = "GPL-3.0" if it's missing after `[package]` in `Cargo.toml`

And lastly, should it go into `scripts/add_license.sh` or in a separate script?

**Comment by dvdplm** [comment]
Maybe add a script that warns us about it but does not take any action? I can imagine situations where adding a mismatching or wrong license could ending real bad. 
On the other hand, what are the legal implications for adding the wrong license to a the crate meta-data? I.e. if the `LICENSE` is a MIT license, but the crate accidentally got published as GPL3, then which is it? IANAL, but surely the actual `LICENSE` file must be what counts?

**Comment by 5chdn** [comment]
I don't want to review this again. Can we maybe split the changes to the script from the changes to the actual headers?

**Comment by niklasad1** [comment]
@5chdn It is only two scripts that are changed:
1. [scripts/add_license.sh](https://github.com/niklasad1/parity/blob/license-header/update/scripts/add_license.sh)
2. [scripts/remove_duplicate_empty_lines.sh](https://github.com/niklasad1/parity/blob/license-header/update/scripts/remove_duplicate_empty_lines.sh)

Do you want me to revert the changes in them? 

Otherwise if I think this PR should be good to go!

**Comment by 5chdn** [comment]
No, sorry, thanks for pointing them out.

**Comment by 5chdn** [comment]
Let's merge this.

**Comment by niklasad1** [comment]
Rebased!

**Comment by 5chdn** [comment]
Something is wrong with the CI. Why is this job not linked @General-Beck ?

https://gitlab.parity.io/parity/parity/-/jobs/88542

**Comment by niklasad1** [comment]
@5chdn it's good to go know :1st_place_medal:

**Comment by debris** [comment]
merging before there are more cnflicts ;)

**Comment by 5chdn** [comment]
Thanks, Niklas :)

**Pull request merged**